src/cli/subcommands/mpool_cmd.rs

```diff
@@ -1,16 +1,408 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+use std::str::FromStr;
+
+use crate::json::cid::vec::CidJsonVec;
+use crate::json::signed_message::json::SignedMessageJson;
+use crate::message::{Message, SignedMessage};
+use crate::rpc_client::{chain_ops::*, mpool_pending, state_ops::*, wallet_ops::*};
+use crate::shim::{address::Address, econ::TokenAmount};
+
+use ahash::{HashMap, HashSet};
 use clap::Subcommand;
+use fvm_ipld_encoding::Cbor;
+use num::BigInt;
 
-use super::Config;
+use super::{handle_rpc_err, Config};
 
 #[derive(Debug, Subcommand)]
-pub enum MpoolCommands {}
+pub enum MpoolCommands {
+    /// Get pending messages
+    Pending {
+        /// Print pending messages for addresses in local wallet only
+        #[arg(long)]
+        local: bool,
+        /// Only print `CIDs` of messages in output
+        #[arg(long)]
+        cids: bool,
+        /// Return messages to a given address
+        #[arg(long)]
+        to: Option<String>,
+        /// Return messages from a given address
+        #[arg(long)]
+        from: Option<String>,
+    },
+    /// Print mempool stats
+    Stat {
+        /// Number of blocks to look back for minimum `basefee`
+        #[arg(long, default_value = "60")]
+        basefee_lookback: u32,
+        /// Print stats for addresses in local wallet only
+        #[arg(long)]
+        local: bool,
+    },
+}
+
+fn to_addr(value: &Option<String>) -> anyhow::Result<Option<Address>> {
+    Ok(value.as_ref().map(|s| Address::from_str(s)).transpose()?)
```

Comment:
You probably want `StrictAddress` for mainnet/testnet safety.

```suggestion
    Ok(value.as_ref().map(|s| StrictAddress::from_str(s)).transpose()?)
```